samtranslator/model/eventsources/pull.py

```diff
@@ -287,7 +287,28 @@ def get_policy_arn(self):  # type: ignore[no-untyped-def]
         return ArnGenerator.generate_aws_managed_policy_arn("service-role/AWSLambdaMSKExecutionRole")  # type: ignore[no-untyped-call]
 
     def get_policy_statements(self):  # type: ignore[no-untyped-def]
-        return None
+        if self.SourceAccessConfigurations:
+            for conf in self.SourceAccessConfigurations:
+                # Lambda does not support multiple CLIENT_CERTIFICATE_TLS_AUTH configurations
+                if isinstance(conf, dict) and conf.get("Type") == "CLIENT_CERTIFICATE_TLS_AUTH" and conf.get("URI"):
+                    return [
+                        {
+                            "PolicyName": "MSKExecutionRolePolicy",
+                            "PolicyDocument": {
+                                "Statement": [
+                                    {
+                                        "Action": [
+                                            "secretsmanager:GetSecretValue",
+                                        ],
+                                        "Effect": "Allow",
+                                        "Resource": conf.get("URI"),
+                                    }
+                                ]
+                            },
+                        }
+                    ]
+
+            return None
```

Comment:
Technically identical, but a little clearer and mypy/pylint won't complain when the `# type: ignore` is removed:

```suggestion
        return None
```